app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java

```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+
+        // Default values
+        binding.tvRecentCount.setText("0");
+        binding.tvRecentRating.setText("n/a");
+        binding.tvCountCompare.setText("\u2195 n/a");
+        binding.tvRatingCompare.setText("\u2195 n/a");
+        binding.tvCountCompare.setTextColor(white);
+        binding.tvRatingCompare.setTextColor(white);
+
+        // Set current year info
+        List<Entry> recentEntries = new ArrayList<>();
+        if (yearToList.containsKey(currentYear)) {
+            recentEntries = yearToList.get(currentYear);
+            binding.tvRecentCount.setText(String.format(Locale.US, "%d", recentEntries.size()));
+            binding.tvRecentRating.setText(String.format(Locale.US, "%.1f", getAverageRating(recentEntries)));
+        }
+
+        // Set comparisons to previous year
+        if (yearToList.containsKey(prevYear)) {
+            List<Entry> prevEntries = yearToList.get(prevYear);
+            setComparisonText(prevEntries.size(), recentEntries.size(), binding.tvCountCompare);
+            if (!binding.tvRecentRating.getText().equals("n/a")) {
+                setComparisonText(getAverageRating(prevEntries),
+                        Double.parseDouble(binding.tvRecentRating.getText().toString()),
+                        binding.tvRatingCompare);
+            }
+        }
+    }
+
+    /* Activity (Line Chart) - year vs. entry count */
+    public void setChartActivity() {
+        LineChart chart = binding.chartActivity;
+
+        // Create data points
+        List<com.github.mikephil.charting.data.Entry> chartEntries = new ArrayList<>();
+        List<Integer> entryCounts = new ArrayList<>();
+        for (int year = minYear; year <= maxYear; year++) {
+            int entryCount = yearToList.containsKey(year)? yearToList.get(year).size(): 0;
+            chartEntries.add(new com.github.mikephil.charting.data.Entry(year, entryCount));
+            entryCounts.add(entryCount);
+        }
+
+        // Turn the points into a data set (sorted chronologically)
+        chartEntries.sort((e1, e2) -> (Double.compare(e1.getX(), e2.getX())));
+        LineDataSet dataSet = new LineDataSet(chartEntries, "Animes Watched");
+        dataSet.setColor(theme);
+        dataSet.setCircleColor(theme);
+        dataSet.setDrawCircleHole(false);
+        dataSet.setValueTextSize(12f);
+        dataSet.setLineWidth(1.5f);
+        dataSet.setCircleRadius(4f);
+        dataSet.setHighLightColor(theme);
+        dataSet.setDrawHighlightIndicators(false);
+        dataSet.setDrawValues(false);
+        chart.setData(new LineData(dataSet));
+
+        // Customize the chart
+        customizeChart(chart, 0);
+        customizeXAxis(chart.getXAxis());
+        customizeYAxisMain(chart.getAxisLeft());
+        customizeYAxisSecondary(chart.getAxisRight());
+        chart.getXAxis().setAxisMinimum(chartEntries.get(0).getX());
+        chart.getAxisLeft().setLabelCount(Collections.max(entryCounts) + 2, true);
```

Comment:
`chart.getAxisLeft()` seems to be called a few times here. Would it make sense to pull this into a separate variable?